### ISSUE TITLE
Uncaught TypeError: Cannot read property 'JIa' of undefined for cell with Wrap Text and Freeze Pane

### ISSUE THREAD

**Issue by leo-lb** [ONLYOFFICE/DocumentServer#903]
**Do you want to request a *feature* or report a *bug*?**

Bug

**What is the current behavior?**

Some of my users cannot open xlsx documents, some times they can open but not save/edit documents

I could collect two error messages from them (in french):
![Capture d'écran 2020-06-12 16 45 07](https://user-images.githubusercontent.com/44934014/84517497-b1d5ad80-accf-11ea-9135-7a3439e7283e.png)
![Capture d'écran 2020-06-08 13 48 02](https://user-images.githubusercontent.com/44934014/84517766-02e5a180-acd0-11ea-8ae3-c54f6c8ed654.png)

And a console log that is related to the first error message screenshot above:
![Screenshot Capture - 2020-06-12 - 17-06-20](https://user-images.githubusercontent.com/44934014/84517458-a2566480-accf-11ea-891a-250d81ea700d.png)

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem.**

I could not reproduce the issue myself for now, I can access or edit xlsx documents fine

**What is the expected behavior?**

No errors

**Did this work in previous versions of DocumentServer?**

Yes

**DocumentServer version:**

5.5.3

**Operating System:**

macOS

**Browser version:**

Google Chrome 83

**Comment by ShockwaveNN** [comment]
Hi

Is there any errors in `/var/log/onlyoffice`?
But seems this is front-end errors, so I think we will need file in which those errors happens and OS info and browser version from such users

**Comment by leo-lb** [comment]
@ShockwaveNN 

Hello! I couldnt reproduce the issue with any document I was given so it remains mysterious, and because these documents are likely sensitive, it's difficult transfering them to you.

I use docker and there's nothing in logs. There's few database time outs due to routine maintenance but the timestamps do not correspond to when the issues happened for users.

All the debug information came from a single user, so all information in the first issue are for the same user. Other users reported similar issues though, but I can't quite catch them to gather more information now.

Is there a way to figure out what part of the code does this error point to? I realize that the browser tries to execute a minified version.

Thanks a lot.

**Comment by ShockwaveNN** [comment]
> I use docker and there's nothing in logs. 

Seems this is mean, that it's purely JS error, have nothing todo with back-end

> All the debug information came from a single user, so all information in the first issue are for the same user. 

Could you get info which OS and browser this user using?

> Is there a way to figure out what part of the code does this error point to?

Yeah, this is not an easy task, but maybe we can, but first we need user info

**Comment by leo-lb** [comment]
@ShockwaveNN 

Yes I could get the info as written in the first post of the issue:

**Operating System:**

macOS

**Browser version:**

Google Chrome 83

I don't have anything more precise. Users arent very tech savy and I can't assist them on site right now.

**Comment by ShockwaveNN** [comment]
@leo-lb I've contacted with our dev team and we think that this error should be in logs, we duplicate client JS errors in backend logs

Could you make sure that there is no any info in `/var/log/onlyoffice` inside docker container?

**Comment by leo-lb** [comment]
@ShockwaveNN 
With a mix of gzip-ed files, it wasnt easy to search, but here it is! I found something thanks to https://github.com/phiresky/ripgrep-all!

```
[2020-06-08T06:53:46.320] [WARN] nodeJS - Express server starting...
[2020-06-08T06:53:46.324] [WARN] nodeJS - Plugins watch exception (https://nodejs.org/docs/latest/api/fs.html#fs_availability).
[2020-06-08T06:53:46.473] [WARN] nodeJS - Express server listening on port 8000 in production-linux mode
[2020-06-08T07:17:37.755] [WARN] nodeJS - error description: docId = 2233652691 errorId = Other error
[2020-06-08T07:32:16.493] [WARN] nodeJS - error description: docId = 3742853793 errorId = Other error
[2020-06-08T07:32:48.471] [WARN] nodeJS - error description: docId = 3554332356 errorId = Other error
[2020-06-08T08:24:22.845] [WARN] nodeJS - error description: docId = 351825946 errorId = Other error
[2020-06-08T08:24:22.847] [WARN] nodeJS - error description: docId = 4078033474 errorId = Other error
[2020-06-08T08:24:22.848] [WARN] nodeJS - error description: docId = 1212231222 errorId = Other error
[2020-06-08T08:26:12.604] [WARN] nodeJS - error description: docId = 351825946 errorId = Other error
[2020-06-08T08:26:12.606] [WARN] nodeJS - error description: docId = 1212231222 errorId = Other error
[2020-06-08T08:26:12.608] [WARN] nodeJS - error description: docId = 4078033474 errorId = Other error
[2020-06-08T08:45:05.877] [ERROR] nodeJS - changesError: docId = 3596789350 Error: Uncaught TypeError: Cannot read property 'JIa' of undefined Script: https://nextcloud.domain.tld/onlyoffice/5.5.3-39/sdkjs/cell/sdk-all.js Line: 7513:164 userAgent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_4) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/83.0.4103.97 Safari/537.36 platform: MacIntel isLoadFullApi: true isDocumentLoadComplete: true StackTrace: TypeError: Cannot read property 'JIa' of undefined
    at d.BJf (https://nextcloud.domain.tld/onlyoffice/5.5.3-39/sdkjs/cell/sdk-all.js:7513:164)
    at d.KYe (https://nextcloud.domain.tld/onlyoffice/5.5.3-39/sdkjs/cell/sdk-all.js:7552:12)
    at d.HYe (https://nextcloud.domain.tld/onlyoffice/5.5.3-39/sdkjs/cell/sdk-all.js:7542:480)
    at d.lFb (https://nextcloud.domain.tld/onlyoffice/5.5.3-39/sdkjs/cell/sdk-all.js:7559:330)
    at d.heg (https://nextcloud.domain.tld/onlyoffice/5.5.3-39/sdkjs/cell/sdk-all.js:7532:183)
    at d.<anonymous> (https://nextcloud.domain.tld/onlyoffice/5.5.3-39/sdkjs/cell/sdk-all.js:7581:100)
    at e.v_a (https://nextcloud.domain.tld/onlyoffice/5.5.3-39/sdkjs/cell/sdk-all.js:984:106)
    at d.COg (https://nextcloud.domain.tld/onlyoffice/5.5.3-39/sdkjs/cell/sdk-all.js:7581:43)
    at HTMLTextAreaElement.<anonymous> (https://nextcloud.domain.tld/onlyoffice/5.5.3-39/sdkjs/cell/sdk-all.js:7522:236)
[2020-06-08T08:46:35.709] [ERROR] nodeJS - changesError: docId = 3596789350 Error: Uncaught TypeError: Cannot read property 'JIa' of undefined Script: https://nextcloud.domain.tld/onlyoffice/5.5.3-39/sdkjs/cell/sdk-all.js Line: 7513:164 userAgent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_4) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/83.0.4103.97 Safari/537.36 platform: MacIntel isLoadFullApi: true isDocumentLoadComplete: true StackTrace: TypeError: Cannot read property 'JIa' of undefined
    at d.BJf (https://nextcloud.domain.tld/onlyoffice/5.5.3-39/sdkjs/cell/sdk-all.js:7513:164)
    at d.KYe (https://nextcloud.domain.tld/onlyoffice/5.5.3-39/sdkjs/cell/sdk-all.js:7552:12)
    at d.HYe (https://nextcloud.domain.tld/onlyoffice/5.5.3-39/sdkjs/cell/sdk-all.js:7542:480)
    at d.lFb (https://nextcloud.domain.tld/onlyoffice/5.5.3-39/sdkjs/cell/sdk-all.js:7559:330)
    at d.heg (https://nextcloud.domain.tld/onlyoffice/5.5.3-39/sdkjs/cell/sdk-all.js:7532:183)
    at d.<anonymous> (https://nextcloud.domain.tld/onlyoffice/5.5.3-39/sdkjs/cell/sdk-all.js:7581:100)
    at e.v_a (https://nextcloud.domain.tld/onlyoffice/5.5.3-39/sdkjs/cell/sdk-all.js:984:106)
    at d.COg (https://nextcloud.domain.tld/onlyoffice/5.5.3-39/sdkjs/cell/sdk-all.js:7581:43)
    at HTMLTextAreaElement.<anonymous> (https://nextcloud.domain.tld/onlyoffice/5.5.3-39/sdkjs/cell/sdk-all.js:7522:236)
[2020-06-08T09:40:10.091] [WARN] nodeJS - error description: docId = 2233652691 errorId = Other error
[2020-06-08T09:50:37.226] [WARN] nodeJS - error description: docId = 4107759628 errorId = Update Version error
[2020-06-08T10:47:33.210] [WARN] nodeJS - error description: docId = 4107759628 errorId = Update Version error
[2020-06-08T11:20:35.618] [WARN] nodeJS - error description: docId = 4107759628 errorId = Update Version error
[2020-06-08T11:29:57.601] [WARN] nodeJS - error description: docId = 2902120051 errorId = Update Version error
[2020-06-08T11:45:30.560] [ERROR] nodeJS - changesError: docId = 3596789350 Error: Uncaught TypeError: Cannot read property 'JIa' of undefined Script: https://nextcloud.domain.tld/onlyoffice/5.5.3-39/sdkjs/cell/sdk-all.js Line: 7513:164 userAgent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_4) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/83.0.4103.97 Safari/537.36 platform: MacIntel isLoadFullApi: true isDocumentLoadComplete: true StackTrace: TypeError: Cannot read property 'JIa' of undefined
    at d.BJf (https://nextcloud.domain.tld/onlyoffice/5.5.3-39/sdkjs/cell/sdk-all.js:7513:164)
    at d.KYe (https://nextcloud.domain.tld/onlyoffice/5.5.3-39/sdkjs/cell/sdk-all.js:7552:12)
    at d.HYe (https://nextcloud.domain.tld/onlyoffice/5.5.3-39/sdkjs/cell/sdk-all.js:7542:480)
    at d.$$b (https://nextcloud.domain.tld/onlyoffice/5.5.3-39/sdkjs/cell/sdk-all.js:7561:345)
    at d.AXe (https://nextcloud.domain.tld/onlyoffice/5.5.3-39/sdkjs/cell/sdk-all.js:7572:311)
    at Ia.a.NATIVE_EDITOR_ENJINE.Uc.Pia (https://nextcloud.domain.tld/onlyoffice/5.5.3-39/sdkjs/cell/sdk-all.js:7638:315)
    at d.Pia (https://nextcloud.domain.tld/onlyoffice/5.5.3-39/sdkjs/cell/sdk-all.js:12139:286)
    at HTMLTextAreaElement.So.onkeydown (https://nextcloud.domain.tld/onlyoffice/5.5.3-39/sdkjs/cell/sdk-all.js:12123:220)
[2020-06-08T11:46:39.087] [WARN] nodeJS - error description: docId = 4107759628 errorId = Update Version error
[2020-06-08T11:47:52.864] [ERROR] nodeJS - changesError: docId = 7969051 Error: Uncaught TypeError: Cannot read property 'JIa' of undefined Script: https://nextcloud.domain.tld/onlyoffice/5.5.3-39/sdkjs/cell/sdk-all.js Line: 7513:164 userAgent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_4) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/83.0.4103.97 Safari/537.36 platform: MacIntel isLoadFullApi: true isDocumentLoadComplete: true StackTrace: TypeError: Cannot read property 'JIa' of undefined
    at d.BJf (https://nextcloud.domain.tld/onlyoffice/5.5.3-39/sdkjs/cell/sdk-all.js:7513:164)
    at d.KYe (https://nextcloud.domain.tld/onlyoffice/5.5.3-39/sdkjs/cell/sdk-all.js:7552:12)
    at d.HYe (https://nextcloud.domain.tld/onlyoffice/5.5.3-39/sdkjs/cell/sdk-all.js:7542:480)
    at d.lFb (https://nextcloud.domain.tld/onlyoffice/5.5.3-39/sdkjs/cell/sdk-all.js:7559:330)
    at d.heg (https://nextcloud.domain.tld/onlyoffice/5.5.3-39/sdkjs/cell/sdk-all.js:7532:183)
    at d.<anonymous> (https://nextcloud.domain.tld/onlyoffice/5.5.3-39/sdkjs/cell/sdk-all.js:7581:100)
    at e.v_a (https://nextcloud.domain.tld/onlyoffice/5.5.3-39/sdkjs/cell/sdk-all.js:984:106)
    at d.COg (https://nextcloud.domain.tld/onlyoffice/5.5.3-39/sdkjs/cell/sdk-all.js:7581:43)
    at HTMLTextAreaElement.<anonymous> (https://nextcloud.domain.tld/onlyoffice/5.5.3-39/sdkjs/cell/sdk-all.js:7522:236)
[2020-06-08T12:02:17.861] [WARN] nodeJS - error description: docId = 56434071 errorId = Update Version error
[2020-06-08T12:08:14.069] [WARN] nodeJS - error description: docId = 4107759628 errorId = Update Version error
[2020-06-08T12:09:02.107] [WARN] nodeJS - error description: docId = 4107759628 errorId = Update Version error
[2020-06-08T12:27:51.558] [WARN] nodeJS - error description: docId = 1859081661 errorId = Restore error. Document modified.
[2020-06-08T12:27:51.559] [WARN] nodeJS - error description: docId = 1859081661 errorId = Restore error. Document modified.
[2020-06-08T12:30:32.484] [WARN] nodeJS - error description: docId = 1859081661 errorId = Update Version error
[2020-06-08T12:30:32.485] [WARN] nodeJS - error description: docId = 1859081661 errorId = Update Version error
[2020-06-08T15:05:14.182] [WARN] nodeJS - error description: docId = 3596789350 errorId = Update Version error
[2020-06-08T15:06:26.223] [WARN] nodeJS - error description: docId = 1859081661 errorId = Update Version error
[2020-06-08T15:06:26.226] [WARN] nodeJS - error description: docId = 1859081661 errorId = Update Version error
[2020-06-08T15:09:20.512] [WARN] nodeJS - error description: docId = 2233652691 errorId = Other error
[2020-06-08T15:09:23.371] [WARN] nodeJS - error description: docId = 3596789350 errorId = Update Version error
[2020-06-08T16:08:24.040] [WARN] nodeJS - error description: docId = 2288749649 errorId = Update Version error
[2020-06-08T16:19:14.978] [WARN] nodeJS - error description: docId = 1859081661 errorId = Update Version error
[2020-06-08T18:31:07.919] [WARN] nodeJS - error description: docId = 4167382088 errorId = Update Version error
[2020-06-08T21:18:12.990] [WARN] nodeJS - error description: docId = 4167382088 errorId = Update Version error
[2020-06-09T01:18:45.260] [WARN] nodeJS - error description: docId = 4167382088 errorId = Update Version error
[2020-06-09T05:19:16.891] [WARN] nodeJS - error description: docId = 4167382088 errorId = Update Version error
[2020-06-09T06:43:33.704] [WARN] nodeJS - error description: docId = 4167382088 errorId = Update Version error
[2020-06-09T06:44:44.967] [WARN] nodeJS - error description: docId = 1859081661 errorId = Update Version error
[2020-06-09T06:53:05.647] [WARN] nodeJS - error description: docId = 4167382088 errorId = Update Version error
```

**Comment by ShockwaveNN** [comment]
@leo-lb yeah, this is good, I shown this to @trofim24, maybe we'll figure out that happened

**Comment by leo-lb** [comment]
@ShockwaveNN Thank you. I'll tell you if I get more insights as well!

**Comment by ShockwaveNN** [comment]
@leo-lb We dig in this problem, but sadly cannot find any scenario to reproduce this bug.
We think it's related to entering several lines of text in formula edit area, but coul not reproduce it in any case we fought.

So we need some more info, maybe a video from your users

**Comment by leo-lb** [comment]
@ShockwaveNN Trying to get that, I'll be sure to share if I succeed.

**Comment by ShockwaveNN** [comment]
We were able to find your problem, not a very easy scenario:
1. Create and open new Spreadsheet
2. Select last visible row on page (in my case it's 39 row)
3. Freeze panes on this line
4. Select A39
5. Home Tab -> Wrap text
6. Enter "Test Test Test Test Test" in this cell and press enter
7. Press up button to go back to A39 and type "Test" again in that cell

And we got your error
This is issue 45653 in our private bug tracker, thank you for reporting it.

**Comment by leo-lb** [comment]
@ShockwaveNN

Wow alright, congratulations for finding out!

**Comment by leo-lb** [comment]
@ShockwaveNN 

I've obtained a video, do you have a confidential channel for communication?

My email is: lle-bout<|AT|>zaclys.net

**Comment by ShockwaveNN** [comment]
I've received a video and take a look, and there is no much detail in video but we pretty sure reason of bug is same as in scenario I described earlier.
Thank you anyway

**Comment by Rita-Bubnova** [comment]
This issue fixed (https://github.com/ONLYOFFICE/sdkjs/commit/c65824bbeefa161f3fc59bf98c8bb1d83f9e7ca1).

**Comment by ShockwaveNN** [comment]
Seems we forgote to close this issue, fix was released in version v5.6, so this problem should not be actual in latest stable version v6.2
